### DISCUSSION TITLE
Redis Failed To Get Next Job

### DISCUSSION THREAD

**Discussion by yientau** [laravel/framework#50341] [General]
Laravel: 7.14.1
PHP: 7.2.5

`[2024-03-02 11:32:48] production.ERROR: Undefined offset: 1 {"exception":"[object] (ErrorException(code: 0): Undefined offset: 1 at /home/forge/swifthr.io/vendor/laravel/framework/src/Illuminate/Queue/RedisQueue.php:230)
[stacktrace]
#0 /home/forge/swifthr.io/vendor/laravel/framework/src/Illuminate/Queue/RedisQueue.php(230): Illuminate\\Foundation\\Bootstrap\\HandleExceptions->handleError()
#1 /home/forge/swifthr.io/vendor/laravel/framework/src/Illuminate/Queue/RedisQueue.php(169): Illuminate\\Queue\\RedisQueue->retrieveNextJob()
#2 /home/forge/swifthr.io/vendor/laravel/framework/src/Illuminate/Queue/Worker.php(282): Illuminate\\Queue\\RedisQueue->pop()
#3 /home/forge/swifthr.io/vendor/laravel/framework/src/Illuminate/Queue/Worker.php(121): Illuminate\\Queue\\Worker->getNextJob()
#4 /home/forge/swifthr.io/vendor/laravel/framework/src/Illuminate/Queue/Console/WorkCommand.php(112): Illuminate\\Queue\\Worker->daemon()
#5 /home/forge/swifthr.io/vendor/laravel/framework/src/Illuminate/Queue/Console/WorkCommand.php(96): Illuminate\\Queue\\Console\\WorkCommand->runWorker()
#6 /home/forge/swifthr.io/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(36): Illuminate\\Queue\\Console\\WorkCommand->handle()
#7 /home/forge/swifthr.io/vendor/laravel/framework/src/Illuminate/Container/Util.php(37): Illuminate\\Container\\BoundMethod::Illuminate\\Container\\{closure}()
#8 /home/forge/swifthr.io/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(93): Illuminate\\Container\\Util::unwrapIfClosure()
#9 /home/forge/swifthr.io/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(37): Illuminate\\Container\\BoundMethod::callBoundMethod()
#10 /home/forge/swifthr.io/vendor/laravel/framework/src/Illuminate/Container/Container.php(596): Illuminate\\Container\\BoundMethod::call()
#11 /home/forge/swifthr.io/vendor/laravel/framework/src/Illuminate/Console/Command.php(134): Illuminate\\Container\\Container->call()
#12 /home/forge/swifthr.io/vendor/symfony/console/Command/Command.php(256): Illuminate\\Console\\Command->execute()
#13 /home/forge/swifthr.io/vendor/laravel/framework/src/Illuminate/Console/Command.php(121): Symfony\\Component\\Console\\Command\\Command->run()
#14 /home/forge/swifthr.io/vendor/symfony/console/Application.php(971): Illuminate\\Console\\Command->run()
#15 /home/forge/swifthr.io/vendor/symfony/console/Application.php(290): Symfony\\Component\\Console\\Application->doRunCommand()
#16 /home/forge/swifthr.io/vendor/symfony/console/Application.php(166): Symfony\\Component\\Console\\Application->doRun()
#17 /home/forge/swifthr.io/vendor/laravel/framework/src/Illuminate/Console/Application.php(93): Symfony\\Component\\Console\\Application->run()
#18 /home/forge/swifthr.io/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(129): Illuminate\\Console\\Application->run()
#19 /home/forge/swifthr.io/artisan(37): Illuminate\\Foundation\\Console\\Kernel->handle()
#20 {main}
"} `

I have around 100 Job Queues that was managed in Forge. I am using Redis as my queue management server. Once in a while, ALL of my job queues will runs into this error. Once it ran into such errors, all of the job queues will stuck and will never run again until I restarted all of the workers in Forge. What could be the possible issue with this? How can I prevent it?

**Comment by henzeb** [comment]
Did it suddenly happen while not updating laravel or packages? maybe he redis instance is no longer compatible with your very old laravel version?

I really suggest if you have the chance, try upgrading things. at least to laravel 7.30.6. That way we can confirm it's not this very old version.

**Reply by yientau** [reply]
I see. Yes, I think the best way now is to upgrade to latest Laravel version and see if that works.
